ogr/ogrsf_frmts/hana/ogrhanatablelayer.cpp

```diff
@@ -0,0 +1,1762 @@
+        GetFullTableNameQuoted(schemaName_, tableName_).c_str(),
+        JoinStrings(values, ", ").c_str(),
+        QuotedIdentifier(GetFIDColumn()).c_str());
+
+    return dataSource_->PrepareStatement(sql.c_str());
+}
+
+/* -------------------------------------------------------------------- */
+/*                     ResetPreparedStatements()                        */
+/* -------------------------------------------------------------------- */
+
+void OGRHanaTableLayer::ResetPreparedStatements()
+{
+    if (!currentIdentityValueStmt_.isNull())
+        currentIdentityValueStmt_ = nullptr;
+    if (!insertFeatureStmtWithFID_.isNull())
+        insertFeatureStmtWithFID_ = nullptr;
+    if (!insertFeatureStmtWithoutFID_.isNull())
+        insertFeatureStmtWithoutFID_ = nullptr;
+    if (!deleteFeatureStmt_.isNull())
+        deleteFeatureStmt_ = nullptr;
+    if (!updateFeatureStmt_.isNull())
+        updateFeatureStmt_ = nullptr;
+}
+
+/************************************************************************/
+/*                        SetStatementParameters()                      */
+/************************************************************************/
+
+OGRErr OGRHanaTableLayer::SetStatementParameters(
+    odbc::PreparedStatement& statement,
+    OGRFeature* feature,
+    bool newFeature,
+    bool withFID,
+    const char* functionName)
+{
+    OGRHanaFeatureReader featReader(*feature);
+
+    unsigned short paramIndex = 0;
+    int fieldIndex = -1;
+    for (const AttributeColumnDescription& clmDesc : attrColumns_)
+    {
+        if (clmDesc.isFeatureID)
+        {
+            if (!withFID && clmDesc.isAutoIncrement)
+                continue;
+
+            ++paramIndex;
+
+            switch (clmDesc.type)
+            {
+            case odbc::SQLDataTypes::Integer:
+                if (feature->GetFID() == OGRNullFID)
+                    statement.setInt(paramIndex, odbc::Int());
+                else
+                {
+                    if (!CanCastIntBigTo<std ::int32_t>(feature->GetFID()))
+                    {
+                        CPLError(
+                            CE_Failure, CPLE_AppDefined,
+                            "%s: Feature id with value %s cannot "
+                            "be stored in a column of type INTEGER",
+                            functionName,
+                            std::to_string(feature->GetFID()).c_str());
+                        return OGRERR_FAILURE;
+                    }
+
+                    statement.setInt(
+                        paramIndex,
+                        odbc::Int(static_cast<std::int32_t>(feature->GetFID())));
+                }
+                break;
+            case odbc::SQLDataTypes::BigInt:
+                if (feature->GetFID() == OGRNullFID)
+                    statement.setLong(paramIndex, odbc::Long());
+                else
+                {
+                    if (!CanCastIntBigTo<std::int64_t>(feature->GetFID()))
```

Comment:
this shouldn't happen . GIntBig == std::int64_t

Reply:
Removed in c9e8b4bf2dcc5962db78e8063d5c01c8e152b931.